scripts/reqs/idgen.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-2.0-only
+#
+# Copyright (C) 2025 Red Hat, Inc. Alessandro Carminati <acarmina@redhat.com>
+#
+# Scans a given directory tree and adds missing SPDX-Req-ID identifiers
+
+import re
+import sys
+import hashlib
+import os
+
+ID_ERR_PATTERN = r"SPDX-Req-ID:[ \t]*([^ \t].*)$"
+ID_PATTERN = r"SPDX-Req-ID:[ \t]*([0-9a-f]{8})\.([0-9]{3})[ \t]*$"
+ID_NEW_PATTERN = r"SPDX-Req-ID:[ \t]*$"
+COMMENT_START_PATTERN = r"^[ \t]*/\*"
+COMMENT_END_PATTERN = r"\*\/"
+max_id_map = {}
+
+
```

Comment:
In general below it would be good to add comments here and there explaining what the script is doing...